collect_app/src/main/java/org/odk/collect/android/activities/FormFillingActivity.java

```diff
@@ -627,6 +628,12 @@ private void setupViewModels(FormEntryViewModelFactory formEntryViewModelFactory
                 });
             }
         });
+
+        AudioClipViewModel.Factory factory = new AudioClipViewModel.Factory(MediaPlayer::new, scheduler);
+        audioClipViewModel = new ViewModelProvider(this, factory).get(AudioClipViewModel.class);
+        audioClipViewModel.isLoading().observe(this, (isLoading) -> {
+            findViewById(R.id.loading_screen).setVisibility(isLoading ? View.VISIBLE : View.GONE);
```

Comment:
We could definitely introduce a separate loading UI for audio, but I wanted to focus on fixing the ANR itself. We could come back for enhancements like that later if we feel like they're a priority.